src/XMakeTasks/Copy.cs

```diff
@@ -323,20 +332,52 @@ private void LogDiagnostic(string message, params object[] messageArgs)
                     FileUtilities.DeleteNoThrow(destinationFileState.Name);
                 }
 
-                hardLinkCreated = NativeMethods.CreateHardLink(destinationFileState.Name, sourceFileState.Name, IntPtr.Zero /* reserved, must be NULL */);
+                linkCreated = NativeMethods.CreateHardLink(destinationFileState.Name, sourceFileState.Name, IntPtr.Zero /* reserved, must be NULL */);
 
-                if (!hardLinkCreated)
+                if (!linkCreated)
                 {
                     int errorCode = Marshal.GetHRForLastWin32Error();
                     Exception hardLinkException = Marshal.GetExceptionForHR(errorCode);
                     // This is only a message since we don't want warnings when copying to network shares etc.
                     Log.LogMessageFromResources(MessageImportance.Low, "Copy.RetryingAsFileCopy", sourceFileState.Name, destinationFileState.Name, hardLinkException.Message);
                 }
             }
+            else if (UseSymboliclinksIfPossible)
```

Comment:
I was actually thinking of an extract method with some params to disambiguate between hard and sym links